include/dem/grid_motion.h

```diff
@@ -109,14 +111,49 @@ class GridMotion
   move_grid_translational(
     parallel::distributed::Triangulation<dim> &triangulation);
 
+  /**
+   * Carries out translational motion of the triangulation
+   *
+   * @param triangulation Triangulation
+   */
+  void
+  move_grid_due_particles_forces(
+    parallel::distributed::Triangulation<dim> &triangulation);
+
+  /**
+   * Update member information like boundary_forces
+   */
+  void
+  update_parameters_before_motion();
+
+  /**
+   * Update member information like inertia tensor,
+   * center of mass coordinate, etc..
+   */
+  void
+  update_parameters_after_motion();
+
+  /**
+   * Calculate motion_parameters as rotation_angle and shift_vector
+   * members
+   */
+  void
+  calculate_motion_parameters();
+
   // Since the DEM time-step and rotational speed are constant, we calculate the
   // rotation angle at each time-step once in the constructor and define it as a
   // member variable.
-  double rotation_angle;
-
-  unsigned int rotation_axis;
-
-  Tensor<1, dim> shift_vector;
+  Tensor<1,3> rotation_angle;
+  Tensor<1,dim> shift_vector;
+
+  const double boundary_mass;
+  Tensor<1,3> boundary_inertia;
+  Tensor<1,dim> boundary_forces;
+  Tensor<1,dim> boundary_torques;
```

Comment:
Why are those called boundary force/masse/torque. Are they not the information about the triangulation or the container?

Reply:
Yes, i would add triangulation in theirs names, is that enough @blaisb ?